airbyte-integrations/connectors/source-jira/source_jira/client.py

```
@@ -0,0 +1,101 @@
+        "users": {"url": "/users/search", "func": lambda v: v, "params": PARAMS},
+        "resolutions": None,
+    }
+
+    def __init__(self, api_token, domain, email):
+        self.auth = HTTPBasicAuth(email, api_token)
+        self.base_api_url = f"https://{domain}/rest/api/{self.API_VERSION}"
+        super().__init__()
+
+    def lists(self, name, **kwargs):
+        while True:
+            response = requests.get(f"{self.base_api_url}{kwargs['url']}", params=kwargs["params"], auth=self.auth)
+            data = kwargs["func"](response.json())
+            for obj in data:
+                if name == "issue_comments":
+                    for comment in obj["fields"]["comment"]["comments"]:
+                        yield comment
+                else:
+                    yield obj
+            kwargs["params"]["startAt"] += self.DEFAULT_ITEMS_PER_PAGE
+            if len(data) < self.DEFAULT_ITEMS_PER_PAGE:
+                break
+
+    def _enumerate_methods(self) -> Mapping[str, callable]:
+        return {
+            entity: getattr(self, f"get_{entity}", None) or partial(self.lists, name=entity, **value)
+            for entity, value in self.ENTITIES_MAP.items()
+        }
+
+    def health_check(self) -> Tuple[bool, str]:
+        alive = True
+        error_msg = None
+
+        try:
+            response = requests.get(f"{self.base_api_url}/users/search", params={"maxResults": 1, "startAt": 0}, auth=self.auth)
+
+            if response.status_code == 200:
+                try:
+                    json.loads(response.text)
+                except JSONDecodeError:
+                    alive, error_msg = (
+                        False,
+                        "Unable to connect to the Jira API with the provided credentials. Please make sure the input credentials and environment are correct.",
+                    )
+            else:
+                alive, error_msg = False, str(response.text)
```

this also should go to retrieval method, maybe raise an exception there, WDYT?